### ISSUE TITLE
handle more validation via ajv

### ISSUE THREAD

**Issue by boneskull** [appium/appium#15916]
After #15664 (hopefully) lands, we will have the opportunity to validate various bits of user-provided configuration via `ajv`.

There are a few places (off the top of my head) where this would make sense. In relative order of impact:

1.  Extension manifest.  That means stuff that lives in the `appium` property of an extension's `package.json`.  
2. Less important, but related, would be to define the shape of `$APPIUM_HOME/extensions.yaml`.   This file is not intended to be end-user-editable (afaik) but it would be good to check in the case that an end-user monkeys with or otherwise corrupts it. 
3. The contents of the value of `--default-capabilities` arg (`server.defaultCapabilities` object).  This is not validated by the code in #15664 other than "it should be an object".  Note that the schema varies based on drivers used.  Unclear if _any_ validation happens here.
4. Common and extension capabilities during runtime.  wdio provides a very thorough set of types for capabilities for reference.  Ideally, schemas for this exist already (e.g., a schema for W3C-style capabilities), but if they don't, working backwards from the types may be helpful.  This either _depends on_ or _is depended on_ by the previous list item.

The goal of this task should result in a lot of existing handrolled validation code being removed, and avoiding it in the future.

Should this be labeled `enhancement`? It's more of a refactor, though it'll likely positively impact UX and DX.

**Comment by jlipps** [comment]
> Note that the schema varies based on drivers used. Unclear if any validation happens here.

These just get put into caps and I believe pass through the normal driver cap validation flow.

> Ideally, schemas for this exist already (e.g., a schema for W3C-style capabilities)

There are only a handful of official W3C capabilities. The rest are invented by Appium. Drivers should really be the ones defining capabilities and exposing validators/schemas.

**Comment by boneskull** [comment]
Fair enough.  Hopefully drivers won't need to actually _implement_ any validation functions, though.

**Comment by jlipps** [comment]
Right, they just define the minimal schema like they currently do.

**Comment by mykola-mokhnach** [comment]
Added more validation in v4
